project/Build.scala

```diff
@@ -805,6 +805,65 @@ object Build {
       javaOptions := (javaOptions in `scala3-compiler-bootstrapped`).value
     )
 
+  // Scala library compiled by dotty using the latest published sources of the library
+  lazy val `scala3-scala2-library` = project.in(file("scala3-scala2-library")).
+    withCommonSettings(Bootstrapped).
+    dependsOn(dottyCompiler(Bootstrapped) % "provided; compile->runtime; test->test").
+    dependsOn(`scala3-tasty-inspector` % "test->test").
+    settings(commonBootstrappedSettings).
+    settings(
+      javaOptions := (javaOptions in `scala3-compiler-bootstrapped`).value,
+      scalacOptions -= "-Xfatal-warnings",
+      ivyConfigurations += SourceDeps.hide,
+      transitiveClassifiers := Seq("sources"),
+      libraryDependencies +=
+        ("org.scala-lang" % "scala-library" % stdlibVersion(Bootstrapped) % "sourcedeps"),
+      sourceGenerators in Compile += Def.task {
+        val s = streams.value
+        val cacheDir = s.cacheDirectory
+        val trgDir = (sourceManaged in Compile).value / "scala-library-src"
+
+        val report = updateClassifiers.value
+        val scalaLibrarySourcesJar = report.select(
+            configuration = configurationFilter("sourcedeps"),
+            module = (_: ModuleID).name == "scala-library",
+            artifact = artifactFilter(`type` = "src")).headOption.getOrElse {
+          sys.error(s"Could not fetch scala-library sources")
+        }
+
+        FileFunction.cached(cacheDir / s"fetchScalaLibrarySrc",
+            FilesInfo.lastModified, FilesInfo.exists) { dependencies =>
+          s.log.info(s"Unpacking scala-library sources to $trgDir...")
+          if (trgDir.exists)
+            IO.delete(trgDir)
+          IO.createDirectory(trgDir)
+          IO.unzip(scalaLibrarySourcesJar, trgDir)
+
+          ((trgDir ** "*.scala") +++ (trgDir ** "*.java")).get.toSet
+        } (Set(scalaLibrarySourcesJar)).toSeq
+      }.taskValue,
+      sources in Compile ~= (_.filterNot(file =>
+        // sources from https://github.com/scala/scala/tree/2.13.x/src/library-aux
+        file.getPath.endsWith("scala-library-src/scala/Any.scala") ||
+        file.getPath.endsWith("scala-library-src/scala/AnyVal.scala") ||
+        file.getPath.endsWith("scala-library-src/scala/AnyRef.scala") ||
+        file.getPath.endsWith("scala-library-src/scala/Nothing.scala") ||
+        file.getPath.endsWith("scala-library-src/scala/Null.scala") ||
+        file.getPath.endsWith("scala-library-src/scala/Singleton.scala"))),
+      managedClasspath in Test ~= {
+        _.filterNot(file => file.data.getName == s"scala-library-${stdlibVersion(Bootstrapped)}.jar")
+      },
+    )
+
+  lazy val `scala3-scala2-library-tasty-tests` = project.in(file("scala3-scala2-library-tasty-tests")).
```

Comment:
This should have some documentation explaining what it tests and why that needs to be done in a separate project.

Reply:
Added